test/solc/CommandLineParser.cpp

```diff
@@ -420,18 +412,24 @@ BOOST_AUTO_TEST_CASE(invalid_options_input_modes_combinations)
 	map<string, vector<string>> invalidOptionInputModeCombinations = {
 		// TODO: This should eventually contain all options.
 		{"--error-recovery", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
-		{"--experimental-via-ir", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}}
+		{"--experimental-via-ir", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
+		{"--metadata-literal", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
+		{"--model-checker-show-unproved", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
+		{"--model-checker-div-mod-no-slacks", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
+		{"--metadata-hash=swarm", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
 	};
 
 	for (auto const& [optionName, inputModes]: invalidOptionInputModeCombinations)
 		for (string const& inputMode: inputModes)
 		{
 			stringstream sout, serr;
-			vector<string> commandLine = {"solc", optionName, "file", inputMode};
+			size_t pos = optionName.find("=");
+			string optionNameWithoutValue = (pos != string::npos) ? optionName.substr(0, pos) : optionName;
+			vector<string> commandLine = {"solc", optionNameWithoutValue, "file", inputMode};
```

Comment:
Isn't it supposed to have a value on the command line?
```suggestion
			vector<string> commandLine = {"solc", optionName, "file", inputMode};
```

Reply:
If we give the option as it is, it will throw an error as we are checking if the option is added or not in the `CommandLineParser`. That is why changed it to this.

Reply:
Not sure I understand. Are you saying that e.g. `solc --standard-json --metadata-hash=swarm` will report an error and `solc --standard-json --metadata-hash` won't? Isn't the error for `solc --standard-json --metadata-hash=swarm` exactly what should happen?

`solc  --standard-json --metadata-hash` does report an error but it's not the one we expect:
```
the required argument for option '--metadata-hash' is missing
```

Reply:
No that is what I meant. In `CommandLineParser.cpp` we are checking if the option exists or not. There the string is `--metadata-hash` and not `--metadata-hash=swarm`. So in `test/CommandlineParser.cpp` we should be passing only `--metadata-hash` so that the test passes.

Reply:
I'm also asking if this is the right way to test this case. Since the option itself is invalid we need to check its value so I made it like this.

Reply:
If I understand correctly, you're worried that `CommandLineParser` will see `--metadata-hash=swarm` as a single option named `metadata-hash=swarm` rather than `metadata-hash` option with a value of `swarm`? That should not be the case. boost::program_options handles both `--metadata-hash swarm` and `--metadata-hash=swarm` in the same way. So if you call `solc --metadata-hash=swarm`, `m_args.count("metadata-hash")` should still give you 1.

Reply:
@nishant-sachdeva I think this still needs to be changed.

Reply:
Added this change in a later push to the code.